test/testsyncengine.cpp

```diff
@@ -714,6 +726,114 @@ private slots:
 
         QCOMPARE(QFileInfo(fakeFolder.localPath() + "foo").lastModified(), datetime);
     }
+
+    static void setAllPerm(FileInfo *fi, RemotePermissions perm)
+    {
+        fi->permissions = perm;
+        for (auto &subFi : fi->children)
+            setAllPerm(&subFi, perm);
+    }
+
+    // What happens if the source can't be moved or the target can't be created?
+    void testForbiddenMoves()
```

Comment:
move to testpermissions.cpp ?  since testsyncengine is big already, and that's where i would expect it.